src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -60,7 +61,31 @@ object LogicalPlanFingerprint {
 // IndexLogEntry-specific SparkPlan that represents the source plan.
 case class SparkPlan(properties: SparkPlan.Properties) {
   val kind = "Spark"
+
+  override def equals(o: Any): Boolean = o match {
+    case that: SparkPlan =>
+      // When a logical plan is serialized, some of the fields can be 'lazy' (e.g.,
+      // Metadata._hashCode), meaning that the same logical plan can result in different serialized
+      // stings. Thus, the serialized strings are deserialized to logical plans and compared.
+      val isPlanEqual = if (!properties.rawPlan.isEmpty) {
```

Comment:
I will update the tests to put in a valid serialized string instead of doing this check.

Reply:
this is no longer relevant